metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -543,8 +552,22 @@ public RollbackResult deleteAspect(String urn, String aspectName, Map<String, St
         _entityDao.deleteAspect(survivingAspect);
       } else {
         if (isKeyAspect) {
-          // If this is the key aspect, delete the entity entirely.
-          additionalRowsDeleted = _entityDao.deleteUrn(urn);
+          if (hardDelete) {
+            // If this is the key aspect, delete the entity entirely.
+            additionalRowsDeleted = _entityDao.deleteUrn(urn);
+          } else if (entitySpec.hasAspect("status")) {
+            // soft delete by setting status.removed=true (if applicable)
+            final Status statusAspect =  new Status();
+            statusAspect.setRemoved(true);
+            final SystemMetadata systemMetadata = new SystemMetadata();
```

Comment:
We def need a helper functino for this :( Feel like it will be useful elsewhere. Fine for now!